src/main/java/org/broadinstitute/dsm/log/SlackAppender.java

```diff
@@ -0,0 +1,144 @@
+                    this.sendSlackNotification(currentEpoch, String.format("This does NOT look like a job error. " +
+                            "Non-job error reporting is throttled so you will only see 1 per %s minutes.", NON_JOB_DELAY), MSG_TYPE_ERROR);
+                    minEpochForNextError.set(currentEpoch + NON_JOB_DELAY * 60L);
+                }
+            } catch (Exception e) {
+                logger.warn("ErrorNotificationAppender Error: " + ExceptionUtils.getStackTrace(e));
+            }
+        }
+    }
+
+    private void sendSlackNotification(long currentEpoch, String note, String messageType) {
+        SlackMessagePayload payload = getSlackMessagePayload(currentEpoch, note, messageType);
+        HttpRequest httpRequest = HttpRequest.newBuilder()
+                .uri(slackHookUrl)
+                .POST(HttpRequest.BodyPublishers.ofString(new Gson().toJson(payload)))
+                .header(HttpHeaders.CONTENT_TYPE, ContentType.APPLICATION_JSON.getMimeType())
+                .build();
+
+        try {
+            HttpResponse<String> response = httpClient.send(httpRequest, HttpResponse.BodyHandlers.ofString());
+            if (response.statusCode() != 200) {
+                throw new IOException("Could not post " + payload + " to slack.  Hook returned " + response.body());
+            }
+        } catch (IOException | InterruptedException e) {
+            logger.warn("Could not post error message to slack room " + slackChannel + " with hook " + slackHookUrl
+                    + "\n" + ExceptionUtils.getStackTrace(e));
+        }
+    }
+
+    public SlackMessagePayload getSlackMessagePayload(long currentEpoch, String note, String messageType) {
+        SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+        String date = formatter.format(new Date(currentEpoch * 1000L));
+        SlackMessagePayload payload = new SlackMessagePayload(String.format("An error has been detected for *%s*." +
+                " Please go check the backend logs around *%s UTC*. \n" + "%s \n %s",
+                appEnv, date, messageType, note), slackChannel, "DSM", ":nerd_face:");
+        return payload;
+    }
+
+    @Override
+    public void close() {
+
+    }
+
+    @Override
+    public boolean requiresLayout() {
+        return false;
+    }
+
+
+    public static synchronized void configure(String scheduler, String appEnv, URI slackHookUri, String slackChannel) {
+        if (!configured) {
+            SlackAppender.appEnv = appEnv;
+            slackHookUrl = slackHookUri;
+            httpClient = HttpClient.newHttpClient();
+            SlackAppender.slackChannel = slackChannel;
+            schedulerName = scheduler;
+            configured = true;
+        } else {
+            throw new RuntimeException("Configure has already been called for this appender.");
+        }
+    }
+
+    static class SlackMessagePayload {
```

Comment:
I think I would prefer the SlackMessagePayload to be an extra class

Reply:
> I think I would prefer the SlackMessagePayload to be an extra class

Since this class is only used in SlackAppender, we created it as inner class, do you think that maybe later we will use this class somewhere else as well?

Reply:
ok, leave it for now and if we need it somewhere else we can pull it out then